spec/requests/customers_spec.rb

```diff
@@ -0,0 +1,116 @@
+    end
+  end
+  describe "post customers_path with invalid data" do
+    it "does not save a new entry or redirect" do
+      customer_attributes = FactoryBot.attributes_for(:customer)
+      customer_attributes.delete(:first_name)
+      expect { post customers_path, params: {customer: customer_attributes}
+    }.to_not change(Customer, :count)
+      expect(response).to render_template(:new)
+    end
+  end
+  describe "put customer_path with valid data" do
+    it "updates an entry and redirects to the show path for the customer" do
+      customer = FactoryBot.create(:customer)
+      put customer_path(customer.id), params: {customer: {phone: "8888888888"}}
+      customer.reload
+      expect(customer.phone).to eq("8888888888")
+      expect(response).to redirect_to customer_path(id: customer.id)
+    end
+  end
+  describe "put customer_path with invalid data" do
+    it "does not update the customer record or redirect" do
+      customer = FactoryBot.create(:customer)
+      put customer_path(customer.id), params: {customer: {phone: "123"}}
+      customer.reload
+      expect(customer.phone).not_to eq("123")
+      expect(response).to render_template(:edit)
+    end
+  end
+  describe "delete a customer record" do
+    it "deletes a customer record" do
+      customer = FactoryBot.create(:customer)
+      delete customer_path(customer.id), params:{}
+      expect(response).to redirect_to customers_path
+    end
+  end
+end
+#
+# .........DEPRECATION WARNING: Using positional arguments in integration tests h
+# as been deprecated,
+# in favor of keyword arguments, and will be removed in Rails 5.1.
+#
+# Deprecated style:
+# get "/profile", { id: 1 }, { "X-Extra-Header" => "123" }
+#
+# New keyword style:
+# get "/profile", params: { id: 1 }, headers: { "X-Extra-Header" => "123" }
+#  (called from block (3 levels) in <top (required)> at /vagrant/Backend-validatio
+# ns/spec/requests/customers_spec.rb:6)
+# .DEPRECATION WARNING: Using positional arguments in integration tests has been d
+# eprecated,
+# in favor of keyword arguments, and will be removed in Rails 5.1.
+#
+# Deprecated style:
+# get "/profile", { id: 1 }, { "X-Extra-Header" => "123" }
+#
+# New keyword style:
+# get "/profile", params: { id: 1 }, headers: { "X-Extra-Header" => "123" }
+#  (called from block (3 levels) in <top (required)> at /vagrant/Backend-validatio
+# ns/spec/requests/customers_spec.rb:13)
+# .DEPRECATION WARNING: Using positional arguments in integration tests has been d
+# eprecated,
+# in favor of keyword arguments, and will be removed in Rails 5.1.
+#
+# Deprecated style:
+# get "/profile", { id: 1 }, { "X-Extra-Header" => "123" }
+#
+# New keyword style:
+# get "/profile", params: { id: 1 }, headers: { "X-Extra-Header" => "123" }
+#  (called from block (3 levels) in <top (required)> at /vagrant/Backend-validatio
+# ns/spec/requests/customers_spec.rb:17)
+# .**..***
+#
+# Pending: (Failures listed here are expected and do not affect your suite's statu
```

Comment:
This `Pending` message is just telling you that the pending tests won't be run && affect your tests.
These are not error messages, so don't worry about them.